libnavigation-core/src/main/java/com/mapbox/navigation/core/ev/EVRerouteOptionsAdapter.kt

```diff
@@ -0,0 +1,29 @@
+package com.mapbox.navigation.core.ev
+
+import com.google.gson.JsonPrimitive
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.navigation.core.reroute.RerouteOptionsAdapter
+import com.mapbox.navigation.core.routeoptions.isEVRoute
+
+internal class EVRerouteOptionsAdapter(
+    private val evDynamicDataHolder: EVDynamicDataHolder
+) : RerouteOptionsAdapter {
+
+    override fun onRouteOptions(routeOptions: RouteOptions): RouteOptions {
+        if (!routeOptions.isEVRoute()) {
+            return routeOptions
+        }
+        val latestEvData = HashMap(
+            evDynamicDataHolder.currentData(
+                routeOptions.unrecognizedJsonProperties ?: emptyMap()
```

Comment:
While that's true right now, `isEVRoute` is a black box for the adapter, the adapter doesn't know what that extension actually checks. I think it's safer to avoid the assumptions (in case the `isEVRoute` impl changes) and use the compiler's checks instead.